writers/httpforwarder/messages.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package httpforwarder
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mainflux/mainflux/errors"
+	"github.com/mainflux/mainflux/transformers/senml"
+	"github.com/mainflux/mainflux/writers"
+)
+
+const pointName = "messages"
+
+var errSaveMessage = errors.New("failed to send message to host")
+
+var _ writers.MessageRepository = (*httpforwarderRepo)(nil)
+
+type httpforwarderRepo struct {
+	url string
+}
+
+type fields map[string]interface{}
+
+// New returns new HTTP forwarder.
+func New(url string) writers.MessageRepository {
+	return &httpforwarderRepo{
+		url: url,
+	}
+}
+
+func (repo *httpforwarderRepo) Save(messages ...senml.Message) error {
+	msgs := make(map[string][]*fields)
+	for _, msg := range messages {
+		_t := repo.fullTopic(&msg)
+		_fields := repo.fieldsOf(&msg)
+		msgs[_t] = append(msgs[_t], &_fields)
+	}
+
+	for topic, msg := range msgs {
+		data, err := json.Marshal(msg)
+		if err != nil {
+			return errors.Wrap(errSaveMessage, err)
+		}
+
+		url := fmt.Sprintf("%s/%s", strings.TrimRight(repo.url, "/"), topic)
+		req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(data))
+		if err != nil {
+			return errors.Wrap(errSaveMessage, err)
+		}
+
+		req.Header.Add("Content-Type", "application/json")
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			return errors.Wrap(errSaveMessage, err)
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
```

Comment:
Do we need this check: `resp.StatusCode != http.StatusCreated` ?

Reply:
You're right, it's not necessary.